### ISSUE TITLE
String concatenation with a non-string right operand emits nothing ("x" + 1, "x" + current date, ...)

### ISSUE THREAD

**Issue by PaulSnow** [DTRules/DTRules#1251]
Found under #1247.

## What happens
If a string is joined to anything the grammar doesn't read as a string, the concatenation is dropped. The row still compiles, and only the right-hand side of the comparison is left:

| EL (condition) | postfix |
|---|---|
| `"x" + 1 == "x1"` | `"x1" streq` |
| `"x" + 1.5 == "x"` | `"x" streq` |
| `"x" + current date == "x"` | `"x" streq` |
| `"x" + first of accounts == "x"` | `"x" streq` |
| `"x" + copy of accounts == "x"` | `"x" streq` |
| `"x" + s == "x"` (s a string, for comparison) | `"x" s strconcat "x" streq` |

These go through `strConcatInt`, `strConcatFloat`, `strConcatDate`, `strConcatEntity` and `strConcatArray`. PostfixEmitter has no `Visit` method for any of them. The Go ANTLR base visitor's `VisitChildren` is a no-op, so they emit nothing. visitor_pin_test.go allowlists all five as "dead grammar; base strConcat wins parser-side". That holds only when the right-hand operand is an identifier: an identifier parses as `strXmlValue`, so the base `strConcat` takes it. With a literal, a date or an entity/array expression on the right, the specific alternatives are chosen.

The compile sweep does not catch this, because the postfix as a whole is not empty.

## Repro (main, 6a4bc145)
```go
c := NewCompiler()
out, err := c.CompileCondition(`"x" + 1 == "x1"`)
// out == `"x1" streq`, err == nil
```

## Expected
The left operand is converted and concatenated, as `strConcatName` does (`"x" $nm cvs concat`). Runtime behaviour of the current postfix was not checked; it compares whatever is below the right-hand side on the stack.

**Comment by PaulSnow** [comment]
**[builder/opus 22:42Z]** Starting on branch `task/1251-strconcat-nonstring` (from origin/main 6a4bc145). Reproduced at run time through the production path (authoring SDK AddAction/AddCondition -> Save -> RuleSet.LoadFromDirectory -> RSession.Execute), test `pkg/dtrules/authoring/strconcat_nonstring_test.go`, before any fix:

```
set calc.text = "x" + 1      -> postfix `cvs /calc.text xdef`, text = "" (want "x1")
set calc.text = "x" + 1.5    -> text = "" (want "x1.5")
set calc.text = "n=" + (2 + 3) -> text = "" (want "n=5")
condition "x" + 1 == "x1"  -> postfix `"x1" streq`, Execute fails: [Stack Underflow] DataPop: Stack is empty  PostFix: "x1"  ERROR==> s==
condition "x" + current date == "x" + string value of current date -> Stack Underflow at s==
```
Fix plan: emitters for strConcatInt/Float/Date/Entity/Array emitting `<s> <rhs> strconcat` (the op stringifies both operands, as the documented `taxpayer.filing_status + taxpayer.age` form relies on). No grammar change.

**Comment by PaulSnow** [comment]
**[builder/opus 22:46Z]** Done: PR https://github.com/DTRules/DTRules/pull/1272, branch `task/1251-strconcat-nonstring`, commit 18bc713c.

**Correction to my first note.** The `text = ""` results there came from my harness: a column with no condition never fires. The real symptom for the action form is a run-time error: `set calc.text = "x" + 1` -> `[Stack Underflow] DataPop: Stack is empty  PostFix: ERROR==> cvs <== /calc.text xdef`.

**Fix.** strConcatInt/Float/Date/Entity/Array emit `<s> <x> strconcat`.

**Tests.**
- `go test ./pkg/dtrules/authoring -run TestStrConcatNonString`: PASS with the fix. With the emitter change removed, all 5 cases FAIL (3 actions with the Stack Underflow at `cvs`, 2 conditions with the Stack Underflow at `s==`).
- `compiler/el` TestStrConcatNonString_Postfix pins all five forms.
- `make check`: passed.
- Merges cleanly with origin/task/1229-string-array-index and origin/task/1249-locals-defined (git merge-tree).

**Sample projects.** I recompiled every sample row before and after the fix. Exactly one changes: TaxReturn/Calculate_Educator_Expenses action 1, where `add "..." + taxpayer.name + ": $" + (the minimum of ...) to job.audit_trail` was compiling to `job.audit_trail swap addto` with no string built. The stored sample XML still holds that broken postfix. I did not regenerate it.

**Not done.**
- strConcatNull / strConcatInvalid are still unemitted. The issue did not list them and I did not check whether they are reachable.
- I did not verify entity/array concatenation at run time; there is a postfix pin only.

**Comment by PaulSnow** [comment]
**[builder/opus 23:22Z]** Follow-up on PR #1272 (commit 5ae4cd5a). TaxReturn's stored XML is rebuilt with `dtrules build sampleprojects/TaxReturn --from-excel`. Exactly one line changed: Calculate_Educator_Expenses action 1 now builds the audit string. #1269's drift test (TestSampleProjectsMatchTheirBuild, copied from its branch) passes, and `make check` passes. I did not check whether any TaxReturn test data makes that row fire. strConcatNull/Invalid are filed as #1292: they are unreachable, with the parse-tree proof there.
